lib/chef/node/attribute.rb

```diff
@@ -523,12 +523,16 @@ def method_missing(symbol, *args)
           else
             raise NoMethodError, "Undefined method or attribute `#{symbol}' on `node'"
           end
-        elsif symbol.to_s =~ /=$/
-          Chef.log_deprecation %q{method setting of node attributes (node.foo="bar") is deprecated and will be removed in Chef 13, please use bracket syntax (node["foo"]="bar")}
-          key_to_set = symbol.to_s[/^(.+)=$/, 1]
-          self[key_to_set] = (args.length == 1 ? args[0] : args)
+        elsif symbol.to_s =~ /^\w/
```

Comment:
I kind of think we shouldn't try to fix this anymore.  All we're doing is possibly breaking some people that are using dot assignment notation here with something that doesn't match `\w` and we can't give people the magic behavior they desired when they're using `>=` on values that need to be autovivified.  The addition of `Hash#>=` just exposes bad code that never really worked correctly before -- and all this patch does is change that from an error due to hitting `>=` to an NoMethodError so those people get raises either way.